### ISSUE TITLE
Inconsistency in the usage of `q` query parameter between the endpoints

### ISSUE THREAD

**Issue by Desvelao** [wazuh/wazuh#18220]
|Wazuh version|Component|
|---|---|
| 4.6.0 | Wazuh component |

## Description

We found some problems in the usage of `q` query parameter:

- Invalid request:
```
GET /mitre/groups?q=(name=W,name=I)
```
API response:
```
{"title": "Bad Request", "detail": "Query does not match expected format: Parameter 'q' is not valid: '(name=W'", "dapi_errors": {"manager-node": {"error": "Query does not match expected format: Parameter 'q' is not valid: '(name=W'"}}, "error": 1407}
```
API log:
```
2023/08/03 07:41:21 INFO: wazuh-wui 172.24.0.8 "GET /mitre/groups" with parameters {"q": "(name=W,name=I)"} and body {} done in 0.015s: 400
```
According to the https://documentation.wazuh.com/4.4/user-manual/api/queries.html, the request seems to be valid, but we get an error.

Meanwhile:
- Valid request:
```
GET /agents?q=(id=001,id=002)
```
API response:
```
{"data": {"affected_items": [{"os": {"arch": "x86_64", "codename": "Focal Fossa", "major": "20", "minor": "04", "name": "Ubuntu", "platform": "ubuntu", "uname": "Linux |23f6e7ecf943 |6.1.29-1-MANJARO |#1 SMP PREEMPT_DYNAMIC Wed May 17 14:00:55 UTC 2023 |x86_64", "version": "20.04.6 LTS"}, "id": "001", "ip": "172.17.0.2", "lastKeepAlive": "2023-08-03T07:45:13+00:00", "node_name": "manager-node", "registerIP": "any", "mergedSum": "4a8724b20dee0124ff9656783c490c4e", "status": "active", "manager": "wazuh-manager-460-7102", "configSum": "ab73af41699f13fdd81903b5f23d8d00", "dateAdd": "2023-08-03T07:07:19+00:00", "name": "23f6e7ecf943", "version": "Wazuh v4.4.5", "group": ["default"], "group_config_status": "synced"}], "total_affected_items": 1, "total_failed_items": 0, "failed_items": []}, "message": "All selected agents information was returned", "error": 0
```
API log:
```
2023/08/03 07:45:15 INFO: wazuh 172.24.0.1 "GET /agents" with parameters {"q": "(id=001,id=002)"} and body {} done in 0.048s: 200
```

It seems the `q` query parameter for the endpoint `GET /mitre/groups` is not managed in a similar way to the `GET /agents` endpoint. We don't know if this problem could be affecting other endpoints.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by RamosFe** [comment]
## Update

Started researching and reading the code of the classes used for each of the endpoints

**Comment by RamosFe** [comment]
## Update

The `agent` controller uses the `WazuhDBQueryAgents` class to execute the query. This class is children of `WazuhDBQuery` that have all the functionality for parsing and creating the SQL equivalent of the query (the `q` option).

However, in the `mitre` controller, the function that handles the logic of the endpoint `mitre/groups` uses the `process_array`, which does not use a database. The logic of the function that handles the parsing of the query differs from the one in `WazuhDBQuery`.

**Comment by Selutario** [comment]
This should be merged in 4.6.1 but there is no branch for said patch version yet. Moving this to **blocked** until then.

**Comment by Selutario** [comment]
It seems there won't be a v4.6.1, so I'm moving it from `blocked` state to `on hold` and assigning 4.7.1 release version @RamosFe
